src/H5Opline.c

```diff
@@ -302,15 +302,18 @@ H5O__pline_encode(H5F_t H5_ATTR_UNUSED *f, uint8_t *p /*out*/, const void *mesg)
             name        = NULL;
         } /* end if */
         else {
-            H5Z_class2_t *cls; /* Filter class */
-
             /*
              * Get the filter name.  If the pipeline message has a name in it then
              * use that one.  Otherwise try to look up the filter and get the name
              * as it was registered.
              */
-            if (NULL == (name = filter->name) && (cls = H5Z_find(filter->id)))
-                name = cls->name;
+            if (NULL == (name = filter->name)) {
+                H5Z_class2_t *cls; /* Filter class */
+
+                H5Z_find(true, filter->id, &cls);
```

Comment:
No checking returned value?

Reply:
Correct - it's a "speculative" lookup ('try' parameter set to true), which doesn't push errors